src/OrleansProviders/Streams/Common/PooledCache/PooledQueueCache.cs

```diff
@@ -94,13 +99,13 @@ private void SetCursor(Cursor cursor, StreamSequenceToken sequenceToken)
                 cursor.State = CursorStates.Idle;
                 cursor.CurrentBlock = newestBlock;
                 cursor.Index = newestBlock.Value.NewestMessageIndex;
-                cursor.SequenceToken = newestBlock.Value.NewestSequenceToken;
+                cursor.SequenceToken = newestBlock.Value.GetNewestSequenceToken(cacheDataAdapter);
                 return;
             }
 
             // If sequenceToken is too new to be in cache, unset token, and wait for more data.
             TCachedMessage newestMessage = newestBlock.Value.NewestMessage;
-            if (cacheDataAdapter.CompareCachedMessageToSequenceToken(ref newestMessage, sequenceToken) < 0) 
+            if (comparer.Compare(newestMessage, sequenceToken) < 0) 
```

Comment:
Interesting post.  I suspect you were just linking that to share it (not specifically related to this change), but the article did make me want to be more clear about my reasoning, as this was not a knee-jerk decision.

When moving functionality out of the data adapter into an IAsymetricComparer, I modeled the comparer off of IComparer.  IComparer arguments are not usually references, so I concluded it was less intuitive to users for IAsymetricComparer arguments to be references.  I was contrasting readability/maintainability/comprehensibility vs. a possible performance gain that I'd not measured.  I concluded I should postpone trading maintainability for performance until I had evidence to defend that decision.  Load testing is planned, and decisions like this will be revisited at that time.

I would argue an optimization is pre-mature if one is trading something of value (developer time, testability, code maintainability, ..) for performance gains when there is little to no reason (data ideally) defending such a trade.  One can't always write fast, clean, testable, and maintainable code, so if a decision is being made to give up something of value for performance reasons, there should be high confidence that it is necessary to do so.